### DISCUSSION TITLE
Dependency Lookup Warnings for in-repo artifacts

### DISCUSSION THREAD

**Discussion by derkoe** [renovatebot/renovate#20232] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I want to get rid of the "Dependency Lookup Warnings" for Maven projects with references to artifacts in the same repository (using parent relation). Here is the dependency dashboard of an example: https://github.com/porscheinformatik/angular-spring-heroes/issues/250

### What you tried so far.

I have seen that this should have been fixed with this PR https://github.com/renovatebot/renovate/pull/18788 and added the ["workarounds:all" preset](https://github.com/porscheinformatik/angular-spring-heroes/blob/main/renovate.json#L9) to my project but the warning is still there.

**Comment by rarkins** [comment]
Possible to reproduce same problem publicly?

**Comment by derkoe** [comment]
The repo is public https://github.com/porscheinformatik/angular-spring-heroes/

**Reply by rarkins** [reply]
So the affected package is at.porscheinformatik.angular-spring-heroes:heroes-parent

Should it be:
- never treated as a dependency in the first place, or
- treated as a skipped dependency? And if so based on exactly which logic?

Trying to work out where the previous attempt fell short

**Reply by derkoe** [reply]
I do not know what is the difference between not a dependency and a skipped dependency but I can give more context.

In Maven you can use parents to have common settings between different modules of your build. All modules are usually built in one multi-module build (called "reactor" in Maven). 

When building a project with a parent, Maven will try to find the parent pom.xml in the parent folder of the file system ("../pom.xml"). If it does not it will look up the dependency in a repository. You can also customize this local path with the `<relativePath>` tag. 

You can also reference other projects in the same reactor build using a  `<dependency>` tag. In this case most people use a variable (`${project.version}` or `${parent.project.version}`) as a version. I think the handling of this is already correct in Renovate.

---

In my view this would be a good implementation for `<parent>` relationship:
- Check if the parent can be found via the `relativePath` (with a default of `../pom.xml`)
- If the pom.xml is found and matches the version then do not treat it as a dependency

---

Good references for this topic:
- [Understanding Maven’s “relativePath” Tag for a Parent POM](https://www.baeldung.com/maven-relativepath)
- [Guide to Working with Multiple Modules
](https://maven.apache.org/guides/mini/guide-multiple-modules.html)

**Reply by derkoe** [reply]
Looking at the code of https://github.com/renovatebot/renovate/pull/18788 this should actually work in my case but it does not.
